### PR TITLE
feat: resolve #32 migrate translations script from other repo

### LINKED ISSUE #32: Migrate Translations Scripts From Related Projects
From ROI/NI and other related apps there is a package that can import/export the translation files which the application uses to/from JSON/Excel files.

### PR DESCRIPTION
**Pull request by jh3y**
- Migrates `translations-script`s from other projects.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +86,7 @@ yarn ios
 ### Contributors
 
 * @dharding - David J Harding <davidjasonharding@gmail.com>
+* @jh3y - Jhey Tompkins <jhey@jhey.dev>
```
Comment (jasnell) [inline README.md]: This should be done in a separate PR once you've been approved as a member.

File: translations-script/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "translations-script",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (jasnell) [inline translations-script/package.json]: License should be Apache-V2

Comment (jasnell) [reply translations-script/package.json]: Or, remove the license field altoghether here.

File: translations-script/clean.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return;
+  }
+
+  Object.keys(source).forEach(key => {
+    process(source[key], target, output, path === '' ? key : `${path}.${key}`);
+  });
+}
+
+async function cleanLangFile(sourceFile, targetFile) {
+  const srcRaw = fs.readFileSync(`../assets/lang/${sourceFile}.json`);
```
Comment (jasnell) [inline translations-script/clean.js]: Curious... why use sync fs methods here? Why not the promisified fs APIs? e.g.

```js
const srcRaw = await fs.promises.readFile(`../assets/lang/${sourceFile}.json`);
```

**Review by jasnell — CHANGES_REQUESTED** [review]
Left a few comments...

File: translations-script/clean.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    return;
+  }
+
+  Object.keys(source).forEach((key) => {
+    process(source[key], target, output, path === '' ? key : `${path}.${key}`);
+  });
+}
+
+async function cleanLangFile(sourceFile, targetFile) {
+  const srcRaw = fs.promises.readFile(`../assets/lang/${sourceFile}.json`);
```
Comment (jasnell) [inline translations-script/clean.js]:
```suggestion
  const srcRaw = await fs.promises.readFile(`../assets/lang/${sourceFile}.json`);
```
?

File: translations-script/clean.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  );
+  const target = JSON.parse(targetRaw);
+
+  const output = {};
+  process(source, target, output, '');
+
+  fs.writeFileSync(
+    `../assets/lang/${targetFile}.json`,
+    JSON.stringify(output, null, 2)
+  );
```
Comment (jasnell) [inline translations-script/clean.js]:
```suggestion
  return fs.promises.writeFile(
    `../assets/lang/${targetFile}.json`,
    JSON.stringify(output, null, 2)
  );
```

File: translations-script/import.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    _.set(ga, path, row[2]);
+  });
+
+  return [en, ga];
+}
+
+main('input.xlsx').then(([en, ga]) => {
+  fs.writeFileSync('en.json', JSON.stringify(en, null, 2));
+  fs.writeFileSync('ga.json', JSON.stringify(ga, null, 2));
+});
```
Comment (jasnell) [inline translations-script/import.js:22]: extremely minor, but these could also be replaced with `fs.promises.writeFile()` but it's fine without.

**Review by jasnell — APPROVED** [review]

File: translations-script/import.js (hunk 6 above)
Comment (jh3y) [reply translations-script/import.js:22]: I did think that but I wasn't 100% sure 👍 
Thank you!

**Review by ShaunBaker — APPROVED** [review]
